org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/XMLValidator.java

```diff
@@ -400,4 +408,22 @@ private static void updateExternalGrammarLocation(DOMDocument document, SAXParse
 			}
 		}
 	}
+
+	/**
+	 * Report error if the path refered to by the catalog URI is invalid
+	 * 
+	 */
+	private static void checkForInvalidCatalogURI(DOMDocument document, List<Diagnostic> diagnostics) {
+		DiagnosticSeverity severity = DiagnosticSeverity.Error;
+
+		for (CatalogEntry catalogEntry : CatalogUtils.getCatalogEntries(document)) {
+			String path = getResolvedLocation(FilesUtils.removeFileScheme(document.getDocumentURI()),

```

Comment:
This piece of code  comes from XMLCatalogDocumentLinkParticipant. I suggest that you create a static method in XMLÇatalogUtils to use it in your code and in the XMlCatalogDocumentLinkParticpant